docs/developer/ceed.md

```diff
@@ -0,0 +1,206 @@
+The block size of `X` is `3` corresponding to the following prognostic variables:
+
+- Height (`h`),
+- Momentum in x-dir (`hu`), and
+- Momentum in y-dir (`hv`).
+The size of `X` will `6 * 3` where `6` corresponds to number of cells in the mesh. The layout of `X` will be as follows:
+
+```text
+X = [[h0 hu0 hv0] [h1 hu1 hv1] ... [h5 hu5 hv5]]
+```
+
+## SWE physics with libCEED
+
+The libCEED version of RDycore's explicit time-integrator of the SWE solver has two `CeedOperator`:
+
+1. `rdy->ceed_rhs.op_edges` : Computes fluxes across edges and it includes two sub-operators that correspond to:
+    - Internal edges,
+    - Boundary edges
+
+2. `rdy->ceed_rhs.src`: Opeator that computes the source terms including rainfall and terms associated with bed slope and bed friction.
+
+### `CeedOperator` for Internal Edges
+
+RDycore uses first-order finite volume discretization to compute the flux across the edges, which requires values on the left and the right of the edge.
+
+| Internal Edge | Left Cell | Right cell |
+| ------------- | --------- | ---------- |
+| e04           | c00       | c01        |
+| e05           | c01       | c02        |
+| e07           | c00       | c03        |
+| e08           | c01       | c04        |
+| e09           | c02       | c05        |
+| e11           | c03       | c04        |
+| e12           | c04       | c05        |
+
+The steps involved in creating the `CeedOperator` associated with the internal edges are as follows:
+
+- First, create a `CeedQFunction`, `qf`, and add input and output fields.
+The input fields includes geometric attributes associated with the edges and the prognostic variables left and right of the edges.
+The output fields include contribution of fluxes to the cells left and right of the edge and a diagnostic variable that saves fluxes through the edge.
+The pointer to the user-defined function is specified at the time of creation.
+
+| Field name | Size | In/Out | Notes                                                        |
+| ---------- | ---- | ------ | ------------------------------------------------------------ |
+| geom       |  4   | In     | Geometric attrbutes [sn, cn, L_edge/Area_left, L_edge/right] |
+| q_left     |  3   | In     | State left of the edge [h_left, hu_left, hv_left] |
+| q_right    |  3   | In     | State right of the edge [h_right, hu_right, hv_right] |
+| cell_left  |  3   | Out    | Flux contribution to the left cell [f_h f_hu f_hv] * L_edge/Area_left |
+| cell_right |  3   | Out    | Flux contribution to the right cell [f_h f_hu f_hv] * L_edge/Area_right |
+| flux       |  3   | Out    | Flux through the edge [f_h f_hu f_hv] |
+
+- Second, create `CeedElemRestriction` for all the input and output fields of previously created `CeedQFunction`.
+A `CeedElemRistriction` tells libCEED the indices of a `CeedVec` from/to which the values are to extracted/written
```

Comment:
`CeedElemRestriction`, and [CeedVector](https://libceed.org/en/latest/api/CeedVector/)